### PR TITLE
feat: startup catch-up nightly for ephemeral machines

### PR DESCRIPTION
**Pull request by kaieriksen**
## Problem

Users who shut down their PC at end of day miss the 02:00 scheduled nightly run. On restart the next morning, memory never gets promoted, KB notes don't get created, and the pipeline described in AGENTS.md silently breaks.

## Solution

At `phantombot run` startup, check if the last nightly run is more than 24 hours stale (or never ran). If so, fire a catch-up nightly in the background — no await, zero impact on startup time.

### Changes

- **`src/lib/nightly.ts`** — new `shouldRunCatchupNightly(personaDir)` + `CATCHUP_WINDOW_MS` constant (24h)
- **`src/cli/run.ts`** — imports `runNightly` and `shouldRunCatchupNightly`; fires background catch-up if stale
- **`tests/lib-nightly.test.ts`** — 5 test cases: no state, stale, recent, unparseable, boundary

### Design decisions

- Uses strict `>` on the 24h boundary so a nightly that ran at exactly 24h ago is still current
- If `last_run` is unparseable, treats it as stale (safe default)
- Errors in the background catch-up are logged but never bubble to the user
- Reuses existing `runNightly()` and `loadNightlyState()` — no new harness or state paths

### CODE REVIEW

File: src/cli/run.ts [hunk 1]
```diff
@@ -89,6 +92,22 @@ export async function runRun(input: RunInput = {}): Promise<number> {
   );
   out.write("Ctrl-C to stop.\n");
 
+  // Catch-up nightly: if the machine was off during the 02:00 window
```
Comment (andrewagrahamhodges) [inline src/cli/run.ts:95]: 🔵 **Trivial**: Good design choice to fire this in the background without awaiting, keeping the startup experience fast.

File: src/lib/nightly.ts [hunk 2]
```diff
@@ -49,6 +49,30 @@ export function nightlyStatePath(personaDir: string): string {
...
+export const CATCHUP_WINDOW_MS = 24 * 60 * 60 * 1000;
+
+/**
+ * Should a catch-up nightly run at startup?
+ *
+ * Returns `true` when the persona has no record of a previous nightly
+ * run OR the last run was more than {@link CATCHUP_WINDOW_MS} ago.
+ * Designed for users who shut down their machine overnight and miss
+ * the 02:00 scheduled run.
+ */
```
Comment (andrewagrahamhodges) [inline src/lib/nightly.ts:65]: 🔵 **Trivial**: Simple and effective check for the 24-hour window.

File: tests/lib-nightly.test.ts [hunk 3]
```diff
@@ -110,6 +112,47 @@ describe("buildNightlyPrompt", () => {
   });
 });
 
+describe("shouldRunCatchupNightly", () => {
```
Comment (andrewagrahamhodges) [inline tests/lib-nightly.test.ts:115]: 🔵 **Trivial**: Strong test coverage for the various state conditions.

**Review by andrewagrahamhodges — APPROVED** [review]
## Code Review: feat: startup catch-up nightly for ephemeral machines

**Author:** @kaieriksen | **Files:** 3 | **Verdict:** Approve

### Findings

| # | Severity | File | Line | Finding |
|---|----------|------|------|---------|
| 1 | 🔵 Trivial | src/cli/run.ts | 95 | Added background catch-up call |
| 2 | 🔵 Trivial | src/lib/nightly.ts | 65 | logic for checking stale nightly state |
| 3 | 🔵 Trivial | tests/lib-nightly.test.ts | 115 | New test cases for catch-up logic |

### What's Good
- Addresses a high-value edge case for users with non-always-on machines.
- Implementation is non-blocking, ensuring startup performance isn't degraded.
- Comprehensive test coverage for the stale state logic.

<details>
<summary>📋 AI Fix Prompt (copy to your AI editor)</summary>

```
Review and fix the following issues found in PR #100 (feat: startup catch-up nightly for ephemeral machines):

1. [Trivial] src/cli/run.ts:95 — Backgrounding the nightly run is a good design choice.
2. [Trivial] src/lib/nightly.ts:65 — Stale check correctly uses a 24h window.
3. [Trivial] tests/lib-nightly.test.ts:115 — Good test coverage.

Repository: phantomyard/phantombot
Branch: feat/startup-nightly-catchup
Base: main
```
</details>
